crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -85,45 +86,61 @@ pub(crate) struct FormatLeadingAlternateBranchComments<'a> {
 
 impl Format<PyFormatContext<'_>> for FormatLeadingAlternateBranchComments<'_> {
     fn fmt(&self, f: &mut PyFormatter) -> FormatResult<()> {
+        let empty_line_condition = if is_blank_line_after_nested_stub_class_enabled(f.context()) {
+            self.last_node.map_or(false, |preceding| {
+                blank_line_after_nested_stub_class_condition(preceding, None, f)
+            })
+        } else {
+            false
+        };
+
```

Comment:
This helps when the class is at the boundary location in nested statement:

```python
if foo:
    class Nested:
        pass
else:
    pass
```

The `FormatLeadingAlternateBranchComments` will be called before the `else` block formatting. Similarly for other blocks like `match`, `case`, `try .. except ..`, etc.